### PR TITLE
feat: allow dapps to suggest petnames for issuer/brand, instance, installation

### LINKED ISSUE #1277: Dapp can suggest petname for Brand/Issuer
**What is the Problem Being Solved?**

The user will want to be able to add issuers and give them petnames from their UI. However, much of this information can be provided by a dapp, presuming that the user consents and is able to change the petname.

**Description of the Design**
The first step would be to create a wallet API method `suggestIssuer(issuerBoardId, suggestedIssuerPetname)` that can be called from the dapp through a websocket connection. We should leave a TODO to allow the user to approve/change the petname through the UI, but that should not be a requirement to get this started.

Eventually, we will want to identify where the suggestion came from, when the petname is displayed. @erights and @cwebber have done the work on how best to display these "edge names".

**Security Considerations**
The dapp is going to be suggesting petnames. If we are not careful about approving them and do not have UI that indicates where the name originated, the suggested petname could mislead us in the future.

**Test Plan**
Write unit tests in test-lib-wallet.js, create a dapp-encouragement branch which uses the endpoint.

### LINKED ISSUE #1278: Dapp can suggest petname for contract instance and installation (actually, for the ids, the instanceHandle and installationHandle)
**What is the Problem Being Solved?**

See related problem https://github.com/Agoric/agoric-sdk/issues/1277

The user will want to be able to add known contract instances and give them petnames from their UI. However, much of this information can be provided by a dapp, presuming that the user consents and is able to change the petname.

**Description of the Design**
The first step would be to create wallet API methods `suggestInstance(instanceHandleBoardId, suggestedInstancePetname)` and `suggestInstallation(installationHandleBoardId, suggestedInstallationPetname)` that can be called from the dapp through a websocket connection. We should leave a TODO to allow the user to approve/change the petname through the UI, but that should not be a requirement to get this started.

**Security Considerations**
The dapp is going to be suggesting petnames. If we are not careful about approving them and do not have UI that indicates where the name originated, the suggested petname could mislead us in the future.

**Test Plan**
Write unit tests in test-lib-wallet.js, create a dapp-encouragement branch which uses the endpoint.

### PR DESCRIPTION
**Pull request by katelynsills**
Also adds renaming and deletion to lib-dehydrate.

Closes #1277 
Closes #1278

### CODE REVIEW

File: packages/cosmic-swingset/test/unitTests/test-lib-dehydrate.js [hunk 1]
```diff
@@ -32,6 +32,62 @@ test('makeDehydrator', async t => {
...
+      `'to be deleted' present going from val to petname`,
+    );
+    t.deepEquals(
+      instanceHandleMapping.petnameToVal.get('to be deleted'),
+      handle1,
+      `'to be deleted' present going from val to petname`,
+    );
+    instanceHandleMapping.deletePetname('to be deleted');
+    console.log(
+      `ERROR EXPECTED '"petname" not found' >>>>`,
```
Comment (michaelfig) [inline packages/cosmic-swingset/test/unitTests/test-lib-dehydrate.js]: Lint in above line.

**Review by michaelfig — APPROVED** [review]
I didn't see any mechanism to rebroadcast the renamed or deleted petnames to the purse/offer observers.  Maybe at least make an issue or a note of where and how this should happen.

Otherwise, looks like a great step forward!  LGTM.

File: packages/cosmic-swingset/test/unitTests/test-lib-dehydrate.js (hunk 1 above)
Comment (katelynsills) [reply packages/cosmic-swingset/test/unitTests/test-lib-dehydrate.js]: Ah good catch

File: packages/cosmic-swingset/lib/ag-solo/vats/lib-wallet.js [hunk 2]
```diff
@@ -615,10 +634,91 @@ export async function makeWallet({
...
+
+    return acceptPetname(
+      // eslint-disable-next-line no-use-before-define
+      wallet.addInstance,
+      suggestedPetname,
+      instanceHandleBoardId,
+    );
+  }
+
+  function suggestInstallation(suggestedPetname, installationHandleBoardId) {
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply packages/cosmic-swingset/lib/ag-solo/vats/lib-wallet.js]: Thanks!

**Comment by katelynsills** [comment]
> I didn't see any mechanism to rebroadcast the renamed or deleted petnames to the purse/offer observers. Maybe at least make an issue or a note of where and how this should happen.
> 
> Otherwise, looks like a great step forward! LGTM.

That is a good point. I think currently, the only display of the petnames is in purses and offers in the inbox. Re-rending both would do the trick since they would automatically get the latest petnames but that seems a bit expensive. Would you be ok with that until we find something better?

**Comment by michaelfig** [comment]
> > I didn't see any mechanism to rebroadcast the renamed or deleted petnames to the purse/offer observers. Maybe at least make an issue or a note of where and how this should happen.

> Re-[s]ending both would do the trick since they would automatically get the latest petnames but that seems a bit expensive. Would you be ok with that until we find something better?

That would be a good tradeoff of correctness versus efficiency (especially since we're not dealing with zillions of items yet), at least until something better!

**Comment by katelynsills** [comment]
@michaelfig Ok, the inefficient but correct updating of state is in.

**Pull request merged**